crates/swc_ecma_transforms_react/src/jsx/mod.rs

```diff
@@ -596,13 +600,62 @@ where
 
                 self.top_level_node = top_level_node;
 
```

Comment:
I think this code will error if `/* @jsxRuntime automatic */` is used.

Reply:
Because of this, the babel plugin detect attributes named `__source` and `__self` and change those into arguments.

Reply:
I’m sorry for that I did not know about jsx directives.

> Because of this, the babel plugin detect attributes named __source and __self and change those into arguments.

Is this the process below?

https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-react-jsx/src/create-plugin.ts#L502

I am worried if my understanding is correct, but should I fix the code to set `__source` and `__self` props in development mode as usual?

Reply:
Sorry for late response, I missed the email.


> Is this the process below?


Yes.

>  Should I fix the code to set __source and __self props as JSX attribute in development mode as usual?

I think so, as it's what babel does

Reply:
Thank you for your reply and help!

I fixed code like below

- set `__source` and `__self` to props with `jsx_src` and `jsx_self` (as usual)
- with `runtime: automatic` and `development: true` configuration, swc extract `__source` and `__self` from JSX’s attributes and set it as `_jsxDEV`’s arguments.

it looks like there are not test settings to test `react` module(= `jsx` with `jsx_src` and  `jsx_self`) with a specific configuration, so I added `integration` testing directory to test `react` module with a specific configuration.

If there is yet something I missed, I will be happy if you pointed it out.